clients/include/testing_csx2dense.hpp

```
@@ -0,0 +1,396 @@
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(
+        handle, M, N, descr, (const T*)d_csx_val, nullptr, d_csx_col_row_ind, (T*)d_dense_val, LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(
+        handle, M, N, descr, (const T*)d_csx_val, d_csx_row_col_ptr, nullptr, (T*)d_dense_val, LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(handle,
+                       M,
+                       N,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)nullptr,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    //
+    // Testing invalid size on M
+    //
+    status = csx2dense(handle,
+                       -1,
+                       N,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
+    //
+    // Testing invalid size on N
+    //
+    status = csx2dense(handle,
+                       M,
+                       -1,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
```

Should be verify_hipsparse_status_invalid_size?

---

I guess it works here because the pointer one just defaults to invalid value internally but given that we have the verify_hipsparse_status_invalid_size method probably we should use it here?

---

Yes :) fixed.